File: SDKs/trashnothing.com/1.2/typescript/src/sdk/models/operations/sharepost.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class SharePostPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=post_id" })
  postId: string;
}


export class SharePostRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: SharePostPathParams;
}


export class SharePostResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/trashnothing.com/1.2/python/src/sdk/models/operations/share_post.py
from dataclasses import dataclass, field



@dataclass
class SharePostPathParams:
    post_id: str = field(metadata={'path_param': { 'field_name': 'post_id', 'style': 'simple', 'explode': False }})
    

@dataclass
class SharePostRequest:
    path_params: SharePostPathParams = field()
    

@dataclass
class SharePostResponse:
    content_type: str = field()
    status_code: int = field()
    
